File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
  let answer = [];
  let complete = [];

  for (let i = 0; i < progresses.length; i++) {
    if (
      complete.length > 0 &&
      complete[complete.length - 1] >=
        Math.ceil((100 - progresses[i]) / speeds[i])
    ) {
      complete.push(complete[complete.length - 1]);
    } else {
      complete.push(Math.ceil((100 - progresses[i]) / speeds[i]));
    }
  }

  for (let i = 0; i < complete.length; i++) {
    if (i === 0 || complete[i] !== complete[i - 1]) {
      answer.push(1);
    } else {
      answer[answer.length - 1] += 1;
    }
  }

  return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math


def solution(progresses, speeds):
    answer = []
    complete = []

    for progress, speed in zip(progresses, speeds):
        if len(complete) > 0 and complete[-1] >= math.ceil((100 - progress) / speed):
            complete.append(complete[-1])
        else:
            complete.append(math.ceil((100 - progress) / speed))
            
    for i in range(len(complete)):
        if i == 0 or complete[i] != complete[i - 1]:
            answer.append(1)
        else:
            answer[-1] += 1
    return answer